### PR TITLE
Fix some types

### LINKED ISSUE #1908: Incorrect types for CascadeSelectProps and ConfirmDialogProps
Incorrect types in ConfirmDialog.d.ts and CascadeSelect.d.ts 

[x ] bug report
[ ] feature request
[ ] support request

**Current behavior**
Type array does not exist <CascadeSelect.d.ts>
Wrong extends in ConfirmDialog.d.ts

**Expected behavior**
Change to `Array<any>`
Change to `extends Omit<DialogProps, 'onHide'>`

### PR DESCRIPTION
**Pull request by mateusnava**
Fixed #1908

### CODE REVIEW

File: src/components/cascadeselect/CascadeSelect.d.ts [hunk 1]
```diff
@@ -5,11 +5,11 @@ interface CascadeSelectProps {
     style?: object;
     className?: string;
     value?: any;
-    options?: array;
+    options?: Array<any>;
```
Comment (master117) [inline src/components/cascadeselect/CascadeSelect.d.ts:8]: I believe any[] would fit better with the rest of PrimeReact typing style from what I have seen. Array\<any\> is only used in Timeline.d.ts, any[] is used in 36 places across 21 files.

```suggestion
    options?: any[];
```

File: src/components/cascadeselect/CascadeSelect.d.ts [hunk 2]
```diff
@@ -5,11 +5,11 @@ interface CascadeSelectProps {
...
     optionLabel?: string;
     optionValue?: string;
     optionGroupLabel?: string;
-    optionGroupChildren?: array;
+    optionGroupChildren?: Array<any>;
```
Comment (master117) [inline src/components/cascadeselect/CascadeSelect.d.ts:12]: I believe any[] would fit better with the rest of PrimeReact typing style from what I have seen. Array\<any\> is only used in Timeline.d.ts, any[] is used in 36 places across 21 files.

```suggestion
    options?: any[];
```

**Review by master117 — CHANGES_REQUESTED** [review]
Looks good to me, but PrimeReact so far uses any[] instead of Array[] in nearly all places.

**Review by master117 — APPROVED** [review]

**Comment by mertsincan** [comment]
Thanks a lot for your contribution.

Best Regards,

**Pull request merged**